scripts/dependencies.py

```diff
@@ -0,0 +1,255 @@
+                    "sha": version["sha"],
+                    "version": version["version"],
+                }
+                break
+        else:
+            raise Exception(
+                (
+                    "The component '{c_name}' has no version matching the "
+                    "constraints '{cs}'."
+                ).format(c_name=component["name"], cs=constraints)
+            )
+
+        return requirement
+
+
+def requirement_from_branch_constraints(component, constraints):
+    constraint = constraints.pop()
+    assert len(constraints) == 0
+
+    constraint_match = re.match(branch_constraint_regex, constraint)
+    branch_name = constraint_match.group("branch_name")
+    with TemporaryDirectory() as tmp_dir:
+        repo = git.Repo.clone_from(releases_repo, tmp_dir, branch=branch_name)
+        sha = repo.head.commit.hexsha
+
+    requirement = {
+        "name": component["name"],
+        "ref": branch_name,
+        "ref_type": "branch",
+        "repo_url": component["repo_url"],
+        "sha": sha,
+        "version": None
+    }
+
+    return requirement
+
+
+def update_requirements(metadata, component_dir):
+    requirements = {"dependencies": []}
+    for dependency in metadata["dependencies"]:
+        component = load_component(dependency["name"], component_dir)
+        constraints = dependency["constraints"]
+        try:
+            branch_constraints_schema.validate(constraints)
+        except SchemaError:
+            requirement = requirement_from_version_constraints(
+                component,
+                constraints
+            )
+        else:
+            requirement = requirement_from_branch_constraints(
+                component,
+                constraints
+            )
+
+        requirements["dependencies"].append(requirement)
+
+    return component_requirements_schema.validate(requirements)
+
+
+def download_requirements(requirements, dl_base_dir):
+    for requirement in requirements:
+        repo_dir = dl_base_dir + requirement["name"]
+        try:
+            repo = git.Repo(repo_dir)
+        except git.exc.NoSuchPathError:
+            repo = git.Repo.clone_from(requirement["repo_url"], repo_dir)
+        else:
+            repo.remote("origin").fetch()
+
+        repo.head.reference = repo.commit(requirement["sha"])
+        repo.head.reset(index=True, working_tree=True)
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--dependency-dir", default="./")
+    parser.add_argument(
+        "--releases-repo",
+        default="https://github.com/rcbops/releases",
+    )
+
+    subparsers = parser.add_subparsers(dest="subparser")
+    subparsers.required = True
+
+    r_parser = subparsers.add_parser("update-requirements")
+    r_parser.add_argument("--component-dir")
+
+    d_parser = subparsers.add_parser("set-dependency")
+    d_parser.add_argument("--name", required=True)
```

Comment:
As mentioned in IRC, we probably need a way to create a skeleton component_metadata.yml file without any constraints (here name is required).

Reply:
Looks like this is still required.  Do we assume if a component has no dependencies then there's no in-repo metadata/requirements files?